src/cli/features.rs

```diff
@@ -0,0 +1,134 @@
+//! Engine feature management.
+
+use cargo;
+
+use std::collections::BTreeMap;
+use toml::{Array, encode_str, Parser, Table, Value};
+
+/// Switches engine features on and off through the `Cargo.toml` manifest and
+/// modifies the `resources` file structure accordingly.
+///
+/// TODO: Should this structure handle file structure modification for features
+/// or should the `Project` structure do it instead?
+///
+/// TODO: Should `Features` or `Project` handle file/folder creation at all?
+/// Perhaps the `Add` and `Remove` subcommands should handle it instead?
+pub struct Features {
+    /// Entire contents of the `Cargo.toml` manifest.
+    manifest: Table,
+    /// The dependencies table.
+    dep_list: Table,
+    /// The `amethyst` entry under dependencies.
+    amethyst: Table,
+}
+
+impl Features {
+    /// Parses the dependencies table in a Cargo manifest.
+    ///
+    /// TODO: Don't like how manifest is loaded straight from the current
+    /// directory instead of being able to specify a path somehow.
+    ///
+    /// TODO: Should this structure be made dependent on the `Project` struct
+    /// and require a borrowed `Project` in the constructor? If so, a lot of the
+    /// error handling can be removed and pushed to `Project` instead, and it
+    /// would be safe to use `unwrap()` for most cases here.
+    pub fn new() -> Result<Features, &'static str> {
+        use std::fs::File;
+        use std::io::Read;
+
+        let mut file = try!(File::open("Cargo.toml").map_err(|_| "Couldn't open Cargo.toml"));
+        let mut text = String::new();
+        try!(file.read_to_string(&mut text).map_err(|_| "Cargo.toml is not a TOML file."));
+
+        let toml = Parser::new(&text).parse().expect("Cargo.toml is invalid.");
```

Comment:
We should route these into actual errors instead of using the expect function to panic. Maybe create a FeaturesError.